azure/ConsiderationsForServiceDesign.md

```diff
@@ -0,0 +1,99 @@
+
+The whole purpose of a preview to address feedback by improving abstractions, naming, relationships, API operations, and so on. It is OK to make breaking changes during a preview to improve the experience now so that it is sustainable long term.
+
+The API should be designed so that developers can use their language of choice to access the service.
+
+> :no_entry: **DO** restrict integer values to the range of +/- 2<sup>53</sup>, since JavaScript native types cannot represent integers outside this range.
+
+### Focus on Hero Scenarios
+It is important to realize that writing an API is, in many cases, the easiest part of providing a delightful developer experience. There are a large number of downstream activities for each API, e.g. testing, documentation, client libraries, examples, blog posts, videos, and supporting customers in perpetuity. In fact, implementing an API is of miniscule cost compared to all the other downstream activities. 
+
+*For this reason, it is <b>much better</b> to ship with fewer features and only add new features over time as required by customers.*
+
+Focusing on hero scenarios reduces development, support, and maintenance costs; enables teams to align and reach consensus faster; and accelerates the time to delivery. A telltale sign of a service that has not focused on hero scenarios is "API drift," where endpoints are inconsistent, incomplete, or juxtaposed to one another. 
+
+> :white_check_mark: **DO** define "hero scenarios" first including abstractions, naming, relationships, and then define the API describing the operations required
+> 
+> :white_check_mark: **DO** provide example code demonstrating the "Hero Scenarios"
+> 
+> :no_entry: **DO NOT** proactively add APIs for speculative features customers might want 
+
+> :white_check_mark: **DO** consider how your abstractions will be represented in different high-level languages.
+
+> :white_check_mark: **DO** develop code examples in at least one dynamically typed language (for example, Python or JavaScript) and one statically typed language (for example, Java or C#) to illustrate your abstractions and high-level language representations.
+
+### Start with your API Definition
+Understanding how your service is used and defining its model and interaction patterns--its API--should be one of the earliest activities a service team undertakes. It reflects the abstractions & naming decisions and makes it easy for developers to implement the hero scenarios.  
+
+> :white_check_mark: **DO** provide an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
```

Comment:
It's not clear where to provide it.

Reply:
I'd like to keep process distinct here (since we are revisiting part of it) but agree we should change the verb from "provide" to "define".